dotnet-engine/Yggdrasil.Engine/YggdrasilEngine.cs

```diff
@@ -17,9 +19,34 @@ public YggdrasilEngine()
         state = FFI.NewEngine();
     }
 
+    public void RegisterCustomStrategies(List<IStrategy> strategies)
+    {
+        customStrategies.RegisterCustomStrategies(strategies);
+    }
+
     public bool ShouldEmitImpressionEvent(string featureName)
     {
-        return FFI.ShouldEmitImpressionEvent(state, featureName);
+        var shouldEmitImpressionEventPtr = FFI.ShouldEmitImpressionEvent(state, featureName);
+        if (shouldEmitImpressionEventPtr == IntPtr.Zero)
```

Comment:
Okay, I see this is a pattern we're using elsewhere but this needs to change in another PR. Nil pointers are a serious problem since the underlying FFI layer can't return those. So something very, very wrong has happened and we need to know about it

I want to say we should draw inspiration from the Java engine here and have a single method can unwrap the incoming FFI JSON bodies and deal with them

Reply:
Yeah I agree, and yeah lets open a new PR with that fix